### ISSUE TITLE
Add a field to the Bootstrap table to expose the purpose of Bootstrap to the consumer.

### ISSUE THREAD

**Issue by tkaven** [zendesk/maxwell#1251]
Add a field to the Bootstrap table to expose the purpose of Bootstrap to the consumer.

Add this key-value pair at the top of the serialized json

In this way, consumers can do relevant processing according to this field

Very practical a function, can achieve it, thank you very much!!

**Comment by tkaven** [comment]
Could you please help to implement this function? Thank you so much!

**Comment by osheroff** [comment]
no, but I'd accept a PR (adding a `comment` field to `maxwell.bootstraps`)

**Comment by tkaven** [comment]
> no, but I'd accept a PR (adding a `comment` field to `maxwell.bootstraps`)

ok, ill try, thank you..

I really need this feature because my clients consume maxwells data directly, and without this field the consumption logic of multiple consumers cannot be distinguished

**Comment by tomcollinsproject** [comment]
I could really use this feature as a way of determining which process kicked off a bootstrap and whether to process the events or not. I'd like to submit a PR. I assume this should be added to the `RowMap` by the `SynchronousBootstrapper`. Would you have any issues with it being a first class property of `RowMap` (not in something like `extraAttributes`)? I use a custom RabbitMQ producer, so once exposed to the `RowMap`, I can take it from there. This property could be added to the various producers over time, however appropriate for that producer type.

Also, while I'm in there, would you have any issue with me changing the `bootstrap.where_clause` from a `varchar(255)` to a `text` type? I've needed that for a bit now and I'm pretty sure I've seen that request from others here.

Thanks!

**Comment by osheroff** [comment]
> a "comment" wouldn't quite represent what I'm doing here, but eh.

you can just call it something opaque like `data` or `info` for all i care.

> Would you have any issues with it being a first class property of RowMap (not in something like extraAttributes)? I use a custom RabbitMQ producer, so once exposed to the RowMap, I can take it from there. 

I'm fine with it being first-class, but please make it a default part of the row serialization so that it's available to all producers.

> Also, while I'm in there, would you have any issue with me changing the bootstrap.where_clause from a varchar(255) to a text type? I've needed that for a bit now and I'm pretty sure I've seen that request from others here.

go right ahead.
